### PR TITLE
Wait for VPC on creation

### PR DESCRIPTION
**Pull request by alevy713**
The VPC is not immediately ready when being created. This can result in errors when setting up a peering connection. If a peering connection is requested befor the VPC is ready, the Timescale VPC will not be found. Now the provider will wait for the VPC to be created as part of the creation workflow.

### CODE REVIEW

File: internal/provider/vpc_resource.go [hunk 1]
```diff
@@ -157,6 +170,41 @@ func (r *vpcResource) Create(ctx context.Context, req resource.CreateRequest, re
...
+			return vpc, vpc.Status, nil
+		},
+	}
+	res, err := conf.WaitForStateContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	vpc, ok := res.(*tsClient.VPC)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type found, expected VPC but got %T", res)
```
Comment (RobAtticus) [inline internal/provider/vpc_resource.go:203]: future improvement: make a dedicated error for unexpected types (this is only the second occurrence so not pressing to do now)

File: internal/provider/vpc_resource.go [hunk 2]
```diff
@@ -157,6 +170,41 @@ func (r *vpcResource) Create(ctx context.Context, req resource.CreateRequest, re
 	}
 }
 
+func (r *vpcResource) waitForVPCReadiness(ctx context.Context, id int64, timeouts timeouts.Value) (*tsClient.VPC, error) {
+	tflog.Trace(ctx, "VPCResource.waitForServiceReadiness")
+
+	defaultTimeout := 45 * time.Minute
```
Comment (RobAtticus) [inline internal/provider/vpc_resource.go:176]: any rhyme or reason to land on 45m?

**Review by RobAtticus — APPROVED** [review]

File: internal/provider/vpc_resource.go (hunk 2 above)
Comment (alevy713) [reply internal/provider/vpc_resource.go:176]: copied this from services, VPC creation should probably be a lot lower. I'll go with 5 minutes to start maybe?

File: internal/provider/vpc_resource.go (hunk 1 above)
Comment (alevy713) [reply internal/provider/vpc_resource.go:203]: true, now im thinking maybe the func could be 

```go
func waitForReadiness[T any](ctx context.Context, id int64, timeouts timeouts.Value, refreshFunc retry.StateRefreshFunc) (*T, error) {
```

**Comment by alevy713** [comment]
added a test to check the VPC status

File: internal/provider/vpc_resource.go (hunk 2 above)
Comment (RobAtticus) [reply internal/provider/vpc_resource.go]: maybe 15 or 30, 45 just seemed like an interesting timeout

**Pull request merged**